### PR TITLE
Combinatorial bid ws

### PR DESCRIPTION
**Pull request by etorres404**

### CODE REVIEW

File: src/main/java/socket/BidWS.java [hunk 1]
```diff
@@ -353,6 +379,155 @@ protected void onAuctionBid(BodyWS body) {
...
+                dbBid = bidDAO.create(newBid);
+            } catch (DAOException e) {
+                Logger.error("Create bid", newBid.toString(), e.toString());
+                sender.reply(session, body, BodyWSCommon.internalServerError());
+                return;
+            }
+        }
+
+        // In this case we protect the bid amount
+        dbBid.amount = 0.0;
```
Comment (zurfyx) [inline src/main/java/socket/BidWS.java]: If you're doing this here  you also need to restrict all servlets transactions listings.

File: src/main/java/socket/BidWS.java [hunk 2]
```diff
@@ -353,6 +379,155 @@ protected void onAuctionBid(BodyWS body) {
...
+
+        List<Bid> unsafeBidsList = new ArrayList<>(Arrays.asList(unsafeBids));
+
+        // 1. All bids must have the same auction ID and amount.
+        double commonAmount = Math.floor(unsafeBids[0].amount * 100) / 100;
+        int commonAuctionId = unsafeBids[0].auctionId;
+        Set<Integer> differentGoods = new HashSet<>();
+        for (Bid unsafeBid: unsafeBidsList) {
+            unsafeBid.amount = Math.floor(unsafeBid.amount * 100) / 100;
+            if (unsafeBid.amount <= 0.1) {
```
Comment (zurfyx) [inline src/main/java/socket/BidWS.java]: Should be 1.0 (Min) in combinatorial, yes?

File: src/main/java/socket/BidWS.java [hunk 3]
```diff
@@ -353,6 +379,155 @@ protected void onAuctionBid(BodyWS body) {
...
+        UserDAO userDAO = UserDAOSQL.getInstance();
+        GoodDAO goodDAO = GoodDAOSQL.getInstance();
+
+        List<Bid> unsafeBidsList = new ArrayList<>(Arrays.asList(unsafeBids));
+
+        // 1. All bids must have the same auction ID and amount.
+        double commonAmount = Math.floor(unsafeBids[0].amount * 100) / 100;
+        int commonAuctionId = unsafeBids[0].auctionId;
+        Set<Integer> differentGoods = new HashSet<>();
+        for (Bid unsafeBid: unsafeBidsList) {
```
Comment (zurfyx) [inline src/main/java/socket/BidWS.java]: For most checks I think you didn't really need the loop.
As long as the first one carried the right information, you could ignore the rest.

File: src/main/java/socket/BidWS.java (hunk 2 above)
Comment (etorres404) [reply src/main/java/socket/BidWS.java]: Yes.

File: src/main/java/socket/BidWS.java (hunk 1 above)
Comment (etorres404) [reply src/main/java/socket/BidWS.java]: True. I will modify this later.

File: src/main/java/socket/BidWS.java (hunk 3 above)
Comment (zurfyx) [reply src/main/java/socket/BidWS.java]: @etorres404 yeah, it's fine

**Pull request merged**